src/oaklib/datamodels/lexical_index.yaml

```diff
@@ -76,6 +76,8 @@ classes:
       pipeline:
         range: LexicalTransformationPipeline
         multivalued: true
+      synonymized:
```

Comment:
Why do we need this? Synonymization should be just like any other transform

Reply:
This flag helps me with updating `mapping_justification`. The value for the `pipeline_name` is broad where in it would be associated with terms which had no change after `synonymizer` rules were implemented. For e.g.: `foo bar` will remain the same as compared to `foo bar (foo bar)` (which gets synonymized to `foo bar`). Now when a transformation happens, this flag (`synonymized`) is set to `True` and further down the line the `mapping_justification` can be updated based on this flag for a particular term.

Reply:
ok let's keep it this way for now but later on we will refactor. We should have a more general mechanism for this. E.g. if a string is affected by XTransformation, we add that to a list for that string in the index (currently we can retrieve which transformations were executed, but not which ones had an effect)